Controller/Midpage/Cancel.php

```diff
@@ -0,0 +1,43 @@
+<?php
+namespace Billink\Billink\Controller\Midpage;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Payment\Gateway\Data\PaymentDataObjectInterface;
+
+class Cancel extends AbstractAction
+{
+    /**
+     * @return
+     */
+    public function execute()
+    {
+        $resultRedirect = $this->redirectFactory->create();
+
+        $transactionOrder = (string)$this->getOrderFromTransaction();
+        if (!$transactionOrder) {
+            $resultRedirect->setPath('checkout/cart');
+            return $resultRedirect;
+        }
+
+        $params = ['_secure' => true];
+        try {
+            $order = $this->checkoutSession->getLastRealOrder();
```

Comment:
Is this order already loaded, or are we triggering a load. if triggering, might as well skip this and immediatly do ->loadOrderByIncrementId

Reply:
No, this is intended.
Transaction Id above is from the request, while this one validate is we are on the same session as this transaction or IF magento session died and we should finish the order as it is.